### PR TITLE
Feat/shlink filters

### PR DESCRIPTION
**Pull request by godchiSymbionix**
Implemented share link filters:
Expired - if expiry date defined and less than today
Active - Active if not expired and active = true
Inactive - If active = false

## Summary by CodeRabbit

- **New Features**
	- Added a new "active" property to the share link schema, indicating if a share link is currently active.
	- Introduced support for filtering share links by their status (expired, active, inactive) in the API.
	- Enhanced dynamic response handling for the share link API.
	- Implemented a validation function for status parameters to ensure correct values are used.

- **Bug Fixes**
	- Improved error handling for invalid status parameters in the API.

- **Documentation**
	- Updated documentation to reflect new query parameters and response behaviors in the API.

### CODE REVIEW

**Review by medchedli — CHANGES_REQUESTED** [review]
Could you please regenerate the Swagger docs to reflect the recent changes

File: src/app/api/v1/share-links/route.ts [hunk 1]
```diff
@@ -69,12 +82,21 @@ export async function POST(request: Request) {
...
+  }
+
   try {
+    unstable_noStore();
     const { id } = await getUserProfile(request);
 
     logger.info(`Getting all share links by user with user id: ${id}`);
 
-    const newShlink = await getSHLinkUseCase({ repo }, { user_id: id });
+    const newShlink = await getSHLinkUseCase({ repo }, { user_id: id, status:status });
```
Comment (jacob-khoza-symb) [inline src/app/api/v1/share-links/route.ts:99]: just have status

File: src/app/api/v1/share-links/route.ts [hunk 2]
```diff
@@ -69,12 +82,21 @@ export async function POST(request: Request) {
  *                 $ref: '#/components/schemas/SHLinkMini'
  */
 export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url)
+  let status: string | null = searchParams.get('status')?.toLowerCase() || null
+
+  const validStatuses = ['expired', 'active', 'inactive'];
```
Comment (jacob-khoza-symb) [inline src/app/api/v1/share-links/route.ts:88]: move the validation logic to a place other than the controller

File: src/mappers/shlink-mapper.test.ts [hunk 3]
```diff
@@ -211,6 +212,7 @@ describe('mapModelToMiniDto', () => {
       passwordRequired: true,
       name: 'name',
       expiryDate: date,
+      active:true,
```
Comment (jacob-khoza-symb) [inline src/mappers/shlink-mapper.test.ts:215]: fix the spacing, run yarn lint:fix

**Review by medchedli — APPROVED** [review]

File: src/app/api/route.ts [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
-const logger = new LogHandler(__dirname)
+const logger = new LogHandler(__dirname);
 
 export async function GET(request) {
-    logger.info("API connected successfully");
-    return NextResponse.json({ message: "API Health Check" }, { status: 200 });
+  unstable_noStore();
+  logger.info('API connected successfully');
+  return NextResponse.json(
+    { message: 'API Health Check', env: process.env.POSTGRES_PRISMA_URL },
```
Comment (jacob-khoza-symb) [inline src/app/api/route.ts:12]: please remove the env property

File: src/app/utils/error-handler.ts [hunk 5]
```diff
@@ -21,6 +21,11 @@ export function handleApiValidationError(error: Error, logger:LogHandler) {
       { error: BAD_REQUEST, detail: error },
       { status: 422 },
     );
+  }else if (error instanceof ParameterValidationError) {
+    return NextResponse.json(
+      {error: BAD_REQUEST, detail:error},
```
Comment (jacob-khoza-symb) [inline src/app/utils/error-handler.ts:26]:
```suggestion
      { error: BAD_REQUEST, detail: error },
```

File: src/app/api/v1/share-links/route.ts [hunk 6]
```diff
@@ -69,12 +83,20 @@ export async function POST(request: Request) {
  *                 $ref: '#/components/schemas/SHLinkMini'
  */
 export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url)
+  let status: string | null = searchParams.get('status')?.toLowerCase() || null
+
   try {
+    unstable_noStore();
+
+    if(status){
```
Comment (jacob-khoza-symb) [inline src/app/api/v1/share-links/route.ts:92]: remove the if statement

File: src/app/api/v1/share-links/route.ts [hunk 7]
```diff
@@ -69,12 +83,20 @@ export async function POST(request: Request) {
...
+      validateSHLinkStatusParameter({status});
```
Comment (jacob-khoza-symb) [inline src/app/api/v1/share-links/route.ts:93]: pass parameters to the function

File: src/app/utils/validate.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+      super(message);
+      this.name = 'ParameterValidationError';
+    }
+ }
+
+ interface SHLinkParameters {
+    status?: string;
+  }
+  
+
```
Comment (jacob-khoza-symb) [inline src/app/utils/validate.ts:17]: inconsistent spacing?

File: src/mappers/shlink-mapper.test.ts [hunk 9]
```diff
@@ -172,6 +172,7 @@ describe('mapModelToMiniDto', () => {
       passwordRequired: true,
       url: 'http://localhost:3000/viewer#shlink:/eyJsYWJlbCI6Im5hbWUiLCJ1cmwiOiJodHRwOi8vbG9jYWxob3N0OjMwMDAvYXBpL3YxL3NoYXJlLWxpbmtzL2xpbmstaWQiLCJmbGFnIjoiUCJ9',
       name: 'name',
+      active:true,
```
Comment (jacob-khoza-symb) [inline src/mappers/shlink-mapper.test.ts:175]: make spacing consistent

File: src/mappers/shlink-mapper.test.ts [hunk 10]
```diff
@@ -258,6 +260,7 @@ describe('mapModelToMiniDto', () => {
       name: 'name',
       url: 'http://localhost:3000/viewer#shlink:/eyJsYWJlbCI6Im5hbWUiLCJ1cmwiOiJodHRwOi8vbG9jYWxob3N0OjMwMDAvYXBpL3YxL3NoYXJlLWxpbmtzL2xpbmstaWQiLCJmbGFnIjoiUCJ9',
       expiryDate: date,
+      active:true,
```
Comment (jacob-khoza-symb) [inline src/mappers/shlink-mapper.test.ts:263]: make spacing consistent

**Review by jacob-khoza-symb — APPROVED** [review]
please fix this
image

**Pull request merged**